algo/src/main/java/com/hartwig/serve/sources/ckb/CkbExtractor.java

```diff
@@ -130,174 +133,133 @@ private ExtractionResult getExtractionResult(CkbEntry entry, String gene, String
-                .addEventInterpretations(interpretation)
-                .actionableHotspots(actionableHotspots)
-                .actionableCodons(actionableCodons)
-                .actionableExons(actionableExons)
-                .actionableGenes(actionableGenes)
-                .actionableFusions(actionableFusions)
-                .actionableCharacteristics(actionableCharacteristics)
-                .actionableHLA(actionableHLA);
+    @NotNull
+    private ExtractionResult toExtractionResult(@NotNull String variant, @NotNull String gene, @NotNull EventExtractorOutput output,
+            @NotNull Set<ActionableEntry> actionableEntries, @NotNull EventInterpretation interpretation) {
+
+        List<CodonAnnotation> codons = curateCodons(output.codons());
+
+        ImmutableExtractionResult.Builder extractionResultBuilder = actionableEntries.stream()
+                .map(event -> ImmutableExtractionResult.builder()
+                        .refGenomeVersion(source.refGenomeVersion())
+                        .actionableHotspots(ActionableEventFactory.toActionableHotspots(event, output.hotspots()))
+                        .actionableCodons(ActionableEventFactory.toActionableRanges(event, codons))
+                        .actionableExons(ActionableEventFactory.toActionableRanges(event, output.exons()))
+                        .actionableGenes(Stream.of(output.geneLevel(), output.copyNumber())
+                                .filter(Objects::nonNull)
+                                .map(annotation -> ActionableEventFactory.geneAnnotationToActionableGene(event, annotation))
+                                .collect(Collectors.toSet()))
+                        .actionableFusions(extractNonNullToSet(output.fusionPair(), event, ActionableEventFactory::toActionableFusion))
+                        .actionableCharacteristics(extractNonNullToSet(output.characteristic(),
+                                event,
+                                ActionableEventFactory::toActionableCharacteristic))
+                        .actionableHLA(extractNonNullToSet(output.hla(), event, ActionableEventFactory::toActionableHLa)))
+                .reduce(ImmutableExtractionResult.builder(), (ImmutableExtractionResult.Builder a, ImmutableExtractionResult.Builder b) -> {
+                    ExtractionResult built = b.build();
+                    a.addAllActionableHotspots(built.actionableHotspots());
+                    a.addAllActionableCodons(built.actionableCodons());
+                    a.addAllActionableExons(built.actionableExons());
+                    a.addAllActionableGenes(built.actionableGenes());
+                    a.addAllActionableFusions(built.actionableFusions());
+                    a.addAllActionableCharacteristics(built.actionableCharacteristics());
+                    a.addAllActionableHLA(built.actionableHLA());
+                    return a;
+                });
 
         if (generateKnownEvents) {
```

Comment:
this is the second conditional I was referring to.

Reply:
I'm not sure that inlining is the way to go because this is actually 2 levels deep (`getExtractionResult -> extractAndInterpretEvent -> toExtractionResult`). However, I don't see any reason why we can't inline the annotation at this level and skip the post-extraction annotation step.